akd/src/tree_node.rs

```diff
@@ -505,23 +505,26 @@ pub(crate) enum NodeHashingMode {
     NoLeafEpoch,
 }
 
-pub(crate) fn maybe_node_to_label(input: &Option<TreeNode>) -> NodeLabel {
+pub(crate) fn node_to_label(input: &Option<TreeNode>) -> NodeLabel {
     match input {
         Some(child_state) => child_state.label,
         None => EMPTY_LABEL,
     }
 }
 
-pub(crate) fn maybe_node_to_hash(input: &Option<TreeNode>, hash_mode: NodeHashingMode) -> Digest {
+pub(crate) fn node_to_azks_value(
+    input: &Option<TreeNode>,
+    hash_mode: NodeHashingMode,
+) -> AzksValue {
     match input {
         Some(child_state) => {
-            let mut hash = child_state.hash;
+            let mut value = child_state.hash;
             if let (TreeNodeType::Leaf, NodeHashingMode::WithLeafEpoch) =
                 (child_state.node_type, hash_mode)
             {
-                hash = hash_leaf_with_commitment(hash, child_state.last_epoch);
+                value = AzksValue(hash_leaf_with_commitment(value, child_state.last_epoch).0);
```

Comment:
Semantically, this should be an `AzksValueWithEpoch` I think. If so, we have to make `AzksValue` an enum with two variants `WithEpoch` and `WithoutEpoch` (or something along the lines)

Reply:
True, I haven't decided what the best way to use the distinction is, yet. I mainly wanted to make a distinction between AzksValues that have had their epochs incorporated (which only happens for leaf nodes) and AzksValues that have not had their epochs incorporated (happens for non-leaf nodes).